src/main/java/org/opensearch/sdk/ExtensionSettings.java

```diff
@@ -109,12 +147,20 @@ public static ExtensionSettings readSettingsFromYaml(String extensionSettingsPat
             if (extensionMap == null) {
                 throw new IOException("extension.yml is empty");
             }
+            Map<String, String> otherSettings = new HashMap<>();
+            Set<String> defaultSettings = Set.of("extensionName", "hostAddress", "hostPort", "opensearchAddress", "opensearchPort");
+            for (String settingKey : extensionMap.keySet()) {
+                if (!defaultSettings.contains(settingKey)) {
+                    otherSettings.put(settingKey, extensionMap.get(settingKey).toString());
+                }
+            }
```

Comment:
I am worried we are loading all pieces of config which we might not understand. This could potentially have security problems as we might load unknown code. 
An example vulnerability which was fixed in `yaml.load`[1]
WDYT?

[1] https://github.com/yaml/pyyaml/issues/243

Reply:
That's a good idea, I will check it against a known list of settings keys. I did it this way for dev convenience since there are quite a few new settings introduced in this PR, but you're right that it shouldn't just accept anything as a setting key.